dll/win32/shell32/brfolder.cpp

```diff
@@ -976,30 +1103,15 @@ BrFolder_OnChangeEx(
         case SHCNE_DRIVEADD:
         case SHCNE_MKDIR:
         case SHCNE_CREATE:
-        {
-            // FIXME
-            break;
-        }
         case SHCNE_DRIVEREMOVED:
         case SHCNE_RMDIR:
         case SHCNE_DELETE:
-        {
```

Comment:
Don't stop handling these? Sometimes we get these and [sometimes SHCNE_UPDATEDIR](https://devblogs.microsoft.com/oldnewthing/20210112-00/?p=104707).

Reply:
I didn't stop handling. See carefully the `switch` statement.

Reply:
But you are throwing away useful information. These 3 tell you exactly which item changed. You are ignoring this and refreshing the entire tree instead. The code you delete looks OK to me. Does it not work? And if not, `BrFolder_FindItemByPidl` is broken?

Reply:
There are both file-system desktop and shell namespace desktop.
Is there any good way to update items of both?

Reply:
> Is there any good way to update items of both?

No, I don't think so. SHCNF_PATH means the file-system PIDL is the one that we need to check if it also exists on the desktop and my documents? And what about stuff like OneDrive etc.? Perhaps something to do with PERSIST_FOLDER_TARGET_INFO?

If we decide to just do a full refresh always for now, at least leave a FIXME comment that we should optimize this when we know how to map PIDLs?

Reply:
FIXME comment added.

Reply:
This is `My Documents` virtual folder:
![MyDocumentsVirtualFolder](https://github.com/reactos/reactos/assets/2107452/d2542f0e-ae1a-4492-a49b-cf502e2696a0)

And this is `My Documents` directory:
![MyDocumentsDirectory](https://github.com/reactos/reactos/assets/2107452/fcbdaa29-2242-49cd-a38c-1e55c2e811fb)

They are different.

Reply:
> They are different

Yes, so the question becomes, do we need a `SHELL_GetOtherLocation(LPCITEMIDLIST, LPITEMIDLIST*)` that shell notification handlers call to get the other pidl. And if so, is it just for the desktop and my documents or also for 3rd-party shell extensions and [instance object folders](https://learn.microsoft.com/en-us/previous-versions/ms997573(v=msdn.10)?redirectedfrom=MSDN)?

Reply:
The Desktop virtual folder is a mix of common desktop and personal desktop, and one PIDL may be placed in multiple folders.

Reply:
There are also unknown virtual folders that the user had customized. The mapping of PIDLs cannot be perfectly predicted.

Reply:
> The Desktop virtual folder is a mix of common desktop and personal desktop. One PIDL may be placed in multiple folders.

But they are not the same (on Windows). Meaning, if you have "foo.txt" in the users desktop folder and "foo.txt" in the common desktop folder, the PIDLS (on the virtual desktop) are not the same. The one on the common desktop has a different "PT_*" type (wine pidl.c is broken as usual). On Windows, this PIDL to the common file starts with `xx xx 3A 00`  and the users file is PT_VALUEW (34).

Funny thing, if you call `SHParseDisplayName` on "c:\users\whatever\desktop\file.txt" on Windows, it returns the virtual desktop PIDL `[file.txt] [END]`

Reply:
If the PIDLs are being translated by change notification engine, we should be able to observe that phenomenon of PIDL translation.

Reply:
> If the PIDLs are being translated by change notification engine

https://www.geoffchappell.com/studies/windows/shell/shell32/interfaces/itranslateshellchangenotify.htm